Neos.Flow/Classes/Session/Session.php

```diff
@@ -621,50 +616,16 @@ public function destroy($reason = null)
      * Iterates over all existing sessions and removes their data if the inactivity
      * timeout was reached.
      *
-     * @return integer The number of outdated entries removed
+     * @return integer The number of outdated entries removed or NULL if no such information could be determined
+     * @deprecated will be removed with Flow 9, use SessionManager->collectGarbage
      * @throws \Neos\Cache\Exception
      * @throws NotSupportedByBackendException
      * @api
      */
     public function collectGarbage()
     {
-        if ($this->inactivityTimeout === 0) {
-            return 0;
-        }
-        if ($this->metaDataCache->has('_garbage-collection-running')) {
-            return false;
-        }
-
-        $sessionRemovalCount = 0;
-        $this->metaDataCache->set('_garbage-collection-running', true, [], 120);
-
-        foreach ($this->metaDataCache->getIterator() as $sessionIdentifier => $sessionInfo) {
-            if ($sessionIdentifier === '_garbage-collection-running') {
-                continue;
-            }
-            if (!is_array($sessionInfo)) {
-                $sessionInfo = [
-                    'sessionMetaData' => $sessionInfo,
-                    'lastActivityTimestamp' => 0,
-                    'storageIdentifier' => null
-                ];
-                $this->logger->warning('SESSION INFO INVALID: ' . $sessionIdentifier, $sessionInfo + LogEnvironment::fromMethodName(__METHOD__));
-            }
-            $lastActivitySecondsAgo = $this->now - $sessionInfo['lastActivityTimestamp'];
-            if ($lastActivitySecondsAgo > $this->inactivityTimeout) {
-                if ($sessionInfo['storageIdentifier'] !== null) {
-                    $this->storageCache->flushByTag($sessionInfo['storageIdentifier']);
-                    $sessionRemovalCount++;
-                }
-                $this->metaDataCache->remove($sessionIdentifier);
-            }
-            if ($sessionRemovalCount >= $this->garbageCollectionMaximumPerRun) {
-                break;
-            }
-        }
-
-        $this->metaDataCache->remove('_garbage-collection-running');
-        return $sessionRemovalCount;
+        $result = $this->sessionManager->collectGarbage();
+        return is_null($result) ? 0 : $result;
```

Comment:
i have the feeling we should return `false` here @mficzel ? Otherwise the return type seems to change?